crates/node/src/command.rs

```diff
@@ -197,14 +207,22 @@ pub fn run() -> sc_cli::Result<()> {
                     Some((madara_path.clone() + "/p2p-key.ed25519").into());
                 cli.run.run_cmd.shared_params.base_path = Some((madara_path.clone()).into());
 
-                if cli.run.testnet.is_some() {
-                    copy_chain_spec(madara_path.clone());
+                if cli.run.chain_spec_url.is_some() {
+                    let url = cli.run.chain_spec_url.clone().unwrap();
+                    fetch_chain_spec(
+                        madara_path.clone(),
+                        url.clone()).unwrap();
+                    cli.run.run_cmd.shared_params.chain =
+                        Some(madara_path.clone() + "/chain-specs/" + url.split("/").last().unwrap());
+                }
 
+                if cli.run.testnet.is_some() {
                     match cli.run.testnet {
-                        Some(Testnet::Local) => {
-                            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/local-raw.json");
-                        }
                         Some(Testnet::Sharingan) => {
+                            // even if chain-spec url is specified, overwrite it
+                            fetch_chain_spec(
+                                madara_path.clone(),
+                                "https://chains.madara.zone/sharingan/testnet-sharingan-raw.json".to_string()).unwrap();
```

Comment:
As said, I would feel more confident if we used the URL of our github repository. I don't know this website, who is managing it and how secure it is.

Also easier to keep the spec up to date, should they change one day.

Reply:
madara.zone is managed by keep-starknet-strange, there is a cron job to fetch the chain-spec from the main branch. That being said, I agree with you. I'll change it to the gh url